flipperdevices/flipperzero-firmware
# [FL-2527] Updater: Migrating to new manifest path convention

# What's new

- [FL-2527] Added new update manifest path storage mechanism
- Has backwards compatibility with older versions - can be installed over them as OTA 
- Has NO rollback compatibility as OTA, older versions won't install from this one
- Added UpdateResponse to RPC with detailed status code 

# Verification 

- Install old release
- Put an update package with this version to /ext and /ext/update
- Verify that updating from old release is not possible from /ext and works from /ext/update
- Install update
- Verify that update from new version to itself works from any path
- Verify that installation of older versions is blocked

# Checklist (For Reviewer)



[FL-2527]: https://flipperzero.atlassian.net/browse/FL-2527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
